Week1/homework/world-queries.txt

```diff
@@ -1,40 +1,186 @@
+37 rows in set (0.00 sec)
+4.What's the top 10 countries by SurfaceArea ?
+SELECT name,SurfaceArea FROM country ORDER BY SurfaceArea DESC LIMIT 10;
++--------------------+-------------+
+| name               | SurfaceArea |
++--------------------+-------------+
+| Russian Federation | 17075400.00 |
+| Antarctica         | 13120000.00 |
+| Canada             |  9970610.00 |
+| China              |  9572900.00 |
+| United States      |  9363520.00 |
+| Brazil             |  8547403.00 |
+| Australia          |  7741220.00 |
+| India              |  3287263.00 |
+| Argentina          |  2780400.00 |
+| Kazakstan          |  2724900.00 |
++--------------------+-------------+
+10 rows in set (0.00 sec)
+5.What's the top 10 most populated cities?
+SELECT name,population from city ORDER BY population DESC LIMIT 10;
++-------------------+------------+
+| name              | population |
++-------------------+------------+
+| Mumbai (Bombay)   |   10500000 |
+| Seoul             |    9981619 |
+| São Paulo         |    9968485 |
+| Shanghai          |    9696300 |
+| Jakarta           |    9604900 |
+| Karachi           |    9269265 |
+| Istanbul          |    8787958 |
+| Ciudad de México  |    8591309 |
+| Moscow            |    8389200 |
+| New York          |    8008278 |
++-------------------+------------+
+10 rows in set (0.00 sec)
+6. Select the top 3 countries by population that their names start with the letter ‘P’
+SELECT name,population FROM city WHERE name LIKE 'P%' ORDER BY population DESC LIMIT 3;
++-----------+------------+
+| name      | population |
++-----------+------------+
+| Peking    |    7472000 |
+| Pusan     |    3804522 |
+| Pyongyang |    2484000 |
++-----------+------------+
+3 rows in set (0.00 sec)
+7.What is the population in Asia?
+ SELECT continent,SUM(population) FROM country WHERE continent='Asia' GROUP BY continent;
++-----------+-----------------+
+| continent | SUM(population) |
++-----------+-----------------+
+| Asia      |      3705025700 |
++-----------+-----------------+
+1 row in set (0.08 sec)
+8.Which languages are spoken in the Region of ‘South America’
+ SELECT DISTINCT language FROM countrylanguage JOIN country on countrylanguage.countrycode=country.code WHERE continent='South America';
+language         |
++------------------+
+| Indian Languages |
+| Italian          |
+| Spanish          |
+| Aimará           |
+| Guaraní          |
+| Ketšua           |
+| German           |
+| Japanese         |
+| Portuguese       |
+| Araucan          |
+| Rapa nui         |
+| Arawakan         |
+| Caribbean        |
+| Chibcha          |
+| Creole English   |
+| English          |
+| Creole French    |
+| Hindi            |
+| Sranantonga      |
+| Goajiro          |
+| Warrau           |
++------------------+
+21 rows in set (0.01 sec)
+9.What are the languages spoken on all cities named ‘Barcelona’ (you may need to join 3 tables =) )
+ SELECT DISTINCT countrylanguage.language from countrylanguage
```

Comment:
but this use of distinct is a little more dangerous ! why did you need a distinct here.. maybe you were getting 2 times spanish ? if so.. my inner data scientist wants to know why, how about we print the city name and the country name ?